addons/mail/models/mail_message.py

```diff
@@ -27,7 +28,11 @@ class Message(models.Model):
     @api.model
     def _get_default_from(self):
         if self.env.user.email:
-            return formataddr((self.env.user.name, self.env.user.email))
+            try:
+                formatted_email = format_address((self.env.user.name, self.env.user.email))
+            except:
+                raise UserError(_("Unable to encode sender's email address %s, is it correct?") % self.env.user.email)
```

Comment:
What are the possible failure cases? Can we do something that is more user-friendly than an ugly error popup when this happens? Any safer fallback?

And if we keep an error message, let's please make it more user-friendly. Users don't know about "encoding", and it's their email, not "sender's". They probably need to know how to fix the problem too (e.g. via their user preferences).